### ISSUE TITLE
Incorrect warning when applying VirtualService with similar HTTPRoutes using prefix match

### ISSUE THREAD

**Issue by denniseffing** [istio/istio#36729]
### Bug Description

We have a Spring Boot application that serves its own API endpoints on path `/servicename` and created a simple `VirtualService` to forward the requests using the Istio ingress gateway:

```yaml
  - match:
    - uri:
        prefix: /servicename
    route:
    - destination:
        host: servicename.hc-track.svc.cluster.local
        port:
          number: 9002
```

However, we also want to expose the Spring actuator endpoints served on `/actuator` on `/servicename/actuator`. So we added another route to our `VirtualService` configuration which now looks like this:

```yaml
  - match:
      - uri:
          prefix: /servicename/actuator
    rewrite:
      uri: /actuator
    route:
      - destination:
          host: servicename.hc-track.svc.cluster.local
          port:
            number: 9002
  - match:
    - uri:
        prefix: /servicename
    route:
    - destination:
        host: servicename.hc-track.svc.cluster.local
        port:
          number: 9002
```

Note that we added the second rule _above_ the first rule because its URI match is more specific.

This works like expected, *but* since upgrading to Istio v1.11.2 we get the following warning when applying our `VirtualService`:

> Warning: virtualService rule #0 match #0 of prefix /servicename/actuator is not used (duplicate/overlapping match in rule #0 of prefix /servicename on #1)

The warning doesn't really do anything and seems to be wrong. If I'd guess I'd say that the validation doesn't consider the order of the rules.

### How to reproduce

To reproduce this bug using the bookinfo sample application, I made a small modification to the `bookinfo-gateway.yaml` file which *also* exposes the static files under the path prefix `/static/test` in addition to the `/static` prefix. The new bookinfo `VirtualService` looks like this:

```yaml
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: bookinfo
spec:
  hosts:
  - "*"
  gateways:
  - bookinfo-gateway
  http:
  - match:
    - uri:
        prefix: /static/test
    rewrite:
      uri: /static
    route:
    - destination:
        host: productpage
        port:
          number: 9080
  - match:
    - uri:
        exact: /productpage
    - uri:
        prefix: /static
    - uri:
        exact: /login
    - uri:
        exact: /logout
    - uri:
        prefix: /api/v1/products
    route:
    - destination:
        host: productpage
        port:
          number: 9080
```

Steps:

1. Deploy sample bookinfo application on Kubernetes with Istio
2. Apply `samples/bookinfo/networking/bookinfo-gateway.yaml`
3. Apply update to `VirtualService` from above

**Expected behavior:** Apply works without any issues

**Actual behavior:** Apply works but prints the following warning:

> Warning: virtualService rule #1 match #0 of prefix /static/test is not used (duplicate/overlapping match in rule #1 of prefix /static on #0)
virtualservice.networking.istio.io/bookinfo configured

To test that the modified `VirtualService` config actually works despite this warning, we can run:

1. `curl -s "http://${GATEWAY_URL}/static/test/bootstrap/css/bootstrap.min.css"`
2. `curl -s "http://${GATEWAY_URL}/static/bootstrap/css/bootstrap.min.css"`

### Version

```prose
$ istioctl version
client version: 1.12.1
control plane version: 1.12.1
data plane version: 1.12.1 (7 proxies)
$ kubectl version
Client Version: v1.22.4
Server Version: v1.20.2
```

### Additional Information

Target cluster context: minikube

Running with the following config: 

istio-namespace: istio-system
full-secrets: false
timeout (mins): 30
include: {  }
exclude: { Namespaces: kube-system,kube-public,kube-node-lease,local-path-storage }
end-time: 2022-01-06 12:42:17.997734 +0100 CET

Cluster endpoint: https://192.168.64.148:8443
CLI version:
version.BuildInfo{Version:"1.12.1", GitRevision:"88902a51acfb0383809608ccff169319560f768c", GolangVersion:"go1.17.3", BuildStatus:"Clean", GitTag:"1.12.1"}

The following Istio control plane revisions/versions were found in the cluster:
Revision default:
&version.MeshInfo{
    {
        Component: "pilot",
        Info:      version.BuildInfo{Version:"1.12.1", GitRevision:"88902a51acfb0383809608ccff169319560f768c", GolangVersion:"", BuildStatus:"Clean", GitTag:"1.12.1"},
    },
}

The following proxy revisions/versions were found in the cluster:
Revision default: Versions {1.12.1}

Fetching proxy logs for the following containers:

default/details-v1/details-v1-79f774bdb9-8x2ft/details
default/details-v1/details-v1-79f774bdb9-8x2ft/istio-proxy
default/productpage-v1/productpage-v1-6b746f74dc-fpgsm/istio-proxy
default/productpage-v1/productpage-v1-6b746f74dc-fpgsm/productpage
default/ratings-v1/ratings-v1-b6994bb9-xpgfm/istio-proxy
default/ratings-v1/ratings-v1-b6994bb9-xpgfm/ratings
default/reviews-v1/reviews-v1-545db77b95-r7x8z/istio-proxy
default/reviews-v1/reviews-v1-545db77b95-r7x8z/reviews
default/reviews-v2/reviews-v2-7bf8c9648f-tkkdh/istio-proxy
default/reviews-v2/reviews-v2-7bf8c9648f-tkkdh/reviews
default/reviews-v3/reviews-v3-84779c7bbc-gbxv9/istio-proxy
default/reviews-v3/reviews-v3-84779c7bbc-gbxv9/reviews
istio-ingress/istio-ingressgateway/istio-ingressgateway-844f9d5fdf-7clqt/istio-proxy
istio-system/grafana/grafana-6596ffd498-phsms/grafana
istio-system/istiod/istiod-7845d77c56-qxjj6/discovery
istio-system/jaeger/jaeger-55bd57c78b-gfkth/jaeger
istio-system/kiali/kiali-f68d6fb6-kkjvr/kiali
istio-system/prometheus-server/prometheus-server-57c67b7dc9-qjjvx/prometheus-server
istio-system/prometheus-server/prometheus-server-57c67b7dc9-qjjvx/prometheus-server-configmap-reload

Fetching Istio control plane information from cluster.

Running istio analyze on all namespaces and report as below:
Analysis Report:
Info [IST0118] (Service istio-system/grafana) Port name service (port: 80, targetPort: 3000) doesn't follow the naming convention of Istio port.
Info [IST0118] (Service istio-system/jaeger-agent) Port name config-rest (port: 5778, targetPort: 5778) doesn't follow the naming convention of Istio port.
Info [IST0118] (Service istio-system/jaeger-collector) Port name c-tchan-trft (port: 14267, targetPort: 14267) doesn't follow the naming convention of Istio port.
Info [IST0131] (VirtualService default/bookinfo) VirtualService rule #0 match #0 of prefix /static/test is not used (duplicate/overlapping match in rule #1 of prefix /static on #1).
Creating an archive at /tmp/bug-report.tar.gz.
Cleaning up temporary files in /var/folders/nf/pc5m88b54414kzp5w3s_bygm0000gn/T/bug-report.
Done.

**Comment by bianpengyuan** [comment]
This is fixed by https://github.com/istio/istio/pull/36251

**Comment by bianpengyuan** [comment]
Let me cherry-pick it to release-1.12, should be available in the next 1.12 release.
